### ISSUE TITLE
Using plists

### ISSUE THREAD

**Issue by rafaelcgs10** [nix-community/nix-doom-emacs#344]
If one wants to use plists to improve performance, then these [steps](https://emacs-lsp.github.io/lsp-mode/page/performance/#use-plists-for-deserialization) are necessary.

The steps 2 and 4 could be achieved by running `LSP_USE_PLISTS=true doom sync` again ([source](https://discourse.doomemacs.org/t/using-lsp-use-plists-with-rust-analyzer-stops-updating-diagnostics-on-save/2832)).

I have the `LSP_USE_PLISTS` variable in my environment (used `home.sessionVariables` to set it), but I think this variable should be present during the `doom sync` when running `home-manager switch`, which I don't believe is the case because this environment is something separated (is it?).

The problem is I have lsp mode complaining that it can't understand the messages from the server.
Messages sample:

```
Error processing message (wrong-type-argument hash-table-p (:jsonrpc "2.0" :method "language/progressReport" :params (:id "041d4f5d-2615-47a0-8882-6f7fbf7da54b" :task "Building" :status "0% " :totalWork 1000 :workDone 0 :complete nil))).
Error processing message (wrong-type-argument hash-table-p (:jsonrpc "2.0" :method "language/progressReport" :params (:id "041d4f5d-2615-47a0-8882-6f7fbf7da54b" :task "Building" :status "100% " :totalWork 1000 :workDone 1000 :complete t))). [2 times]
```

My question is: how can I rebuild the packages with `LSP_USE_PLISTS` variable as requests by the instructions?

**Comment by ckiee** [comment]
> which I don't believe is the case because this environment is something separated (is it?).

Yes, all nix builds run in a sandbox. Can read some of the [wiki](https://nixos.wiki)/nix.dev/[manual](https://nixos.org/manual/nix/stable/) on that. You probably want to patch your own version of `nix-straight.el` (also under this org) and stick that envvar in one of the build phases. Good luck; maybe check out [`#doom-emacs:nixos.org`](https://matrix.to/#/#doom-emacs:nixos.org). 

PR possibly okay, but it'd be ugly to generalize without #353 done. Do let me know if you get it though, fast LSP sounds nice.. ^^
